handshake-layer_test.go

```diff
@@ -83,6 +86,13 @@ func TestMessageMarshal(t *testing.T) {
 	assertByteEquals(t, out, tinyMessage)
 }
 
+func newTestHandshakeMessage(t HandshakeType, m []byte) HandshakeMessage {
```

Comment:
Is there any particular reason this got factored out?  Planning something else for this later?

Reply:
Trying to remember, TBH.